python/rascaline/transformer.py

```diff
@@ -0,0 +1,296 @@
+            # those keys that were in the values of the variable moved_keys.
+            # Go through them and create all possible combinations of these
+            # parts.
+            for key in self.tensor_map.keys:
+                for value in self.moved_keys:
+                    clue = [k.copy() for k in key]
+                    clue.extend(value)
+                    new_keys.append(clue)
+            # The keys have been listed in random order, let's arrange them and
+            # store the values in 'final_keys'.
+            indices = []
+            for key in self._old_keys_names:
+                indices.append(names.index(key))
+            ordered_keys = []
+            for el in new_keys:
+                key = [el[i] for i in indices]
+                ordered_keys.append(key)
+            self.final_keys = Labels(
+                names=self._old_keys_names, values=np.array(ordered_keys)
+            )
+
+    def _mover(self, tensor_map):
+        # Internal function that does the transformation of the reference
+        # Tensormap.
+        self._old_keys = tensor_map.keys
+        self._old_keys_names = tensor_map.keys.names
+        tensor_copy = self._copy(tensor_map)
+        if self.transformation is not None:
+            if self.transformation == "keys_to_samples":
+                tensor_copy.keys_to_samples(self.moved_keys)
+            elif self.transformation == "keys_to_properties":
+                tensor_copy.keys_to_properties(self.moved_keys)
+        return tensor_copy
+
+    def properties_selection(self):
+        # This function selects properties according to a preset algorithm
+        # within each 'TensorMap' block
+        blocks = []
+        for _, block in self.tensor_map:
+            mask = self.selector.fit(block.values).get_support()
+            selected_properties = block.properties[mask]
+            blocks.append(
+                TensorBlock(
+                    # Since the resulting 'TensorMap' will then be used as a
+                    # reference, the only thing we are interested in each
+                    # block is the name of the properties.
+                    values=np.empty((1, len(selected_properties))),
+                    samples=Labels.single(),
+                    components=[],
+                    properties=selected_properties,
+                )
+            )
+
+        self.selected_tensor = TensorMap(self.tensor_map.keys, blocks)
+
+    def fit(self, tensor_map):
+        """The fit function tells the transformer which attributes to use when
+        creating new representations.
+
+        Parameters:
+        -----------
+        :param tensor_map: reference TensorMap, with which transformations are
+            carried out and in which properties are selected.
+        """
+        self.tensor_map = self._mover(tensor_map)
+        self.keys_definition()
+        self.properties_selection()
+
+    def transform(self, frames, calculator):
```

Comment:
Should this class take the calculator in the constructor? Given that it will only be usable with one calculator.

Reply:
And this function also needs to handle all other kwargs that could be given to the calculator (mainly `gradients` & `use_native_system`). We could either have them explicitly, or take a `**kwargs` argument, set the corresponding `selected_properties` and pass it to the calculator

Reply:
On the first point, it's an interesting question. So, it is clear that we need to input `frames` (our general task is to get `descriptor` from them as simply as possible). For this, we need a `calculator` one way or another. And now the question is at what point we will feed it as input? We either have to do it in the `fit` function or already in `transform`. In general, the first option is possible (we can pass either `descriptor` and `calculator` or `frames` and `calculator` to `fit`). What do you think would be better?
On the second point, yes, I agree, I'll try to fix that.

Reply:
I would pass the calculator even earlier (in `__init__`), and then both `fit` and `transform` take a set of frames + some options as kwargs